### PR TITLE
Update Simulate Output 

### PR DESCRIPTION
**Pull request by Tom-Szendrey**
# Description
-Rename simulate-ciemss drilldown to keep up with naming.
-Change simulate-ciemss state to include "forecastId" + rename inProgressForecastId
-Simulate node now outputs a dataset ID. This dataset is a temporary asset.

# Note
This is displaying the forecast's results not the output dataset's results.
Im not sure if this is a concern as the dataset created is from the forecast. 
**Please feel free to discuss on this PR.**

### CODE REVIEW

File: packages/client/hmi-client/src/components/workflow/ops/simulate-ciemss/tera-simulate-node-ciemss.vue [hunk 1]
```diff
@@ -126,15 +128,23 @@ Provide a summary in 100 words or less.
 
 	const summaryResponse = await createLLMSummary(prompt);
 
+	const datasetName = `Forecast run ${runId}`;
+	const projectId = useProjects().activeProject.value?.id ?? '';
+	const datasetResult = await createTemporaryDatasetFromSimulationResult(projectId, runId, datasetName);
+	if (!datasetResult) {
+		logger.error('Error creating dataset from simulation result.');
+		return;
+	}
```
Comment (mwdchang) [inline packages/client/hmi-client/src/components/workflow/ops/simulate-ciemss/tera-simulate-node-ciemss.vue:137]: We have something called `createDatasetFromSimulationResult` - I am not familiar with it, but have we looked to see if we can use it or extend the feature to adapt to this use case? Cause at the surface it seems like you are doing very similar things.

https://github.com/DARPA-ASKEM/terarium/blob/412a24f7b5fc14d98dde66d72e21d3de111d6bde/packages/client/hmi-client/src/services/dataset.ts#L273-L276

Comment (Tom-Szendrey) [reply packages/client/hmi-client/src/components/workflow/ops/simulate-ciemss/tera-simulate-node-ciemss.vue:137]: I can add a flag to `createDatasetFromSimulationResult` for temporary 
And change the return type from boolean -> Dataset
as well as update the use of it in `saveDataset` to still have that boolean logic there

Good call give me a moment

File: packages/client/hmi-client/src/services/dataset.ts [hunk 2]
```diff
@@ -252,27 +252,32 @@ async function createNewDatasetFromFile(
...
-): Promise<boolean> {
+	datasetName: string | null,
+	isTemporary?: boolean
+): Promise<Dataset | null> {
+	if (isTemporary === undefined) isTemporary = false;
 	try {
-		const response: AxiosResponse<Response> = await API.post(
-			`/simulations/${simulationId}/add-result-as-dataset-to-project/${projectId}?dataset-name=${datasetName}`
+		const response: AxiosResponse<Dataset> = await API.post(
+			`/simulations/${simulationId}/create-result-as-dataset/${projectId}?dataset-name=${datasetName}&is-temporary=${isTemporary}`
```
Comment (YohannParis) [inline packages/client/hmi-client/src/services/dataset.ts:266]: We have an issue #4228 to remove the `isTemporary`.
So I'm not sure we need to do this, here.

Comment (Tom-Szendrey) [reply packages/client/hmi-client/src/services/dataset.ts:266]: It was my understanding that when we run a forecast with a node we should create a _temporary_ dataset asset.
If the user then wants to save this asset into their project they would use this menu:
image

Which will create a copy of the dataset asset, marked as not temporary, and add this new asset to the project.

Do you have a quick minute to chat about this?

Comment (Tom-Szendrey) [reply packages/client/hmi-client/src/services/dataset.ts]: Thanks Yohann!

**Review by YohannParis — APPROVED** [review]

**Review by mwdchang — APPROVED** [review]
thanks

**Pull request merged**